auth/client/iam/openid4vp.go

```diff
@@ -149,34 +154,63 @@ func (c *OpenID4VPClient) AccessToken(ctx context.Context, code string, verifier
 	return &token, nil
 }
 
-func (c *OpenID4VPClient) CreateAuthorizationRequest(ctx context.Context, requestHolder did.DID, verifier did.DID, scopes string, clientState string) (*url.URL, error) {
+func (c *OpenID4VPClient) CreateAuthorizationRequest(ctx context.Context, client did.DID, server did.DID, modifier RequestModifier) (*url.URL, error) {
 	// we want to make a call according to §4.1.1 of RFC6749, https://www.rfc-editor.org/rfc/rfc6749.html#section-4.1.1
 	// The URL should be listed in the verifier metadata under the "authorization_endpoint" key
 	iamClient := c.newHTTPClient()
-	metadata, err := iamClient.OAuthAuthorizationServerMetadata(ctx, verifier)
+	metadata, err := iamClient.OAuthAuthorizationServerMetadata(ctx, server)
 	if err != nil {
 		return nil, fmt.Errorf("failed to retrieve remote OAuth Authorization Server metadata: %w", err)
 	}
 	if len(metadata.AuthorizationEndpoint) == 0 {
-		return nil, fmt.Errorf("no authorization endpoint found in metadata for %s", verifier)
+		return nil, fmt.Errorf("no authorization endpoint found in metadata for %s", server)
 	}
 	endpoint, err := url.Parse(metadata.AuthorizationEndpoint)
 	if err != nil {
 		return nil, fmt.Errorf("failed to parse authorization endpoint URL: %w", err)
 	}
-	// construct callback URL for wallet
-	callbackURL, err := didweb.DIDToURL(requestHolder)
-	if err != nil {
-		return nil, fmt.Errorf("failed to create callback URL: %w", err)
+	// one default param for both signed and unsigned
+	params := map[string]interface{}{
```

Comment:
why not `map[string]string`? OAuth2 parameters are always strings (as they are encoded in URL query parameters), so no need to support anything else?

Reply:
JWT claims are not exclusively strings. A JAR request is encoded as JWT and does include string arrays as values.

Reply:
fixed with  `singleStringParam`